pkg/promtail/targets/file/tailer.go

```diff
@@ -136,31 +136,38 @@ func (t *tailer) markPositionAndSize() error {
 	t.posAndSizeMtx.Lock()
 	defer t.posAndSizeMtx.Unlock()
 
-	pos, err := t.tail.Tell()
+	size, err := t.tail.Size()
 	if err != nil {
+		// If the file no longer exists, no need to save position information
+		if err == os.ErrNotExist {
+			level.Info(t.logger).Log("msg", "skipping update of position for a file which does not currently exist")
+			return nil
+		}
```

Comment:
This is really the only important change in this function, the diff is screwy because I swapped the order of Tell() and Size() calls.
This is a more graceful way to handle ignoring the size/position for a file which doesn't exist, it leverages the change in the HP tail lib which returns the os.ErrNotExist error